angular/tech-show-frontend/src/app/components/login/login.component.ts

```diff
@@ -7,4 +9,37 @@ import { Component } from '@angular/core';
 })
 export class LoginComponent {
   title = 'Login';
+
+  constructor(private authService: AuthService) {}
+
+  loginForm = new FormGroup({
+    email: new FormControl('', [
+      Validators.required,
+      Validators.pattern('^[a-z0-9._%+-]+@[a-z0-9.-]+.[a-z]{2,4}$')
+    ]),
+    password: new FormControl('', [
+      Validators.required,
+      Validators.minLength(6)
+    ])
+  });
+
+  get email() {
```

Comment:
extra getters?

Reply:
It breaks if I remove them, probably because reactive forms is using it. In the input I have `formControlName="email"`, I'm guessing it uses this getter.